### ISSUE TITLE
Sprint April 1 - April 12, 2024

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#21181]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team Productus Analyticus

**Support hero:** Thomas

Marius of 2 weeks (entire sprint).

## Retro

- Support (Michael)
  - Lots of support
- HogQL trends (Marius)
  - Today ~4 issues to check. Might be less by the time of the meeting.
  - On track to roll out this week.
- HogQL funnels (Thomas)
  - Similar number of issues outstanding.
  - On track to roll out this week.
- HogQL paths -> funnels (Julian)
  - Connected to funnels.
  - Entering testing.

## Hang over items from previous sprint

TODO

## OKRs Q2 2024

- HogQL-based querying
  - Convert the remaining legacy queries to HogQL and release to public (Thomas, Julian, Marius)
    - Trends? Funnels? Paths?
    - Cohorts
  - Remove legacy querying backend (Thomas, Julian)
    - Convert filters to query (insights, notebooks, activity log, experiments)
    - Clean up or rewrite dashboardLogic
  - Missing Product Analytics features (Thomas, Julian)
    - Breakdowns (multiple) in literally everything
    - Make a list based on GitHub issues from customer requests…
  - Missing HogQL features (Tom, Marius)
    - Type system, JSON
    - Missing things when building funnels

- Querying and processing performance (Thomas, Julian)
  - Global performance overview dashboards
    - Insights
    - Exports
    - Cohort recalculations
  - Query request tracing
    - Possibly query runner Python optimizations
    - Exports improvements
  - Identify top 5 query optimizations in terms of impact

- Artificial Hog / Post Intelligence (Michael)
  - Ask a question to get a magical insight (aware of your taxonomy)
  - Figure out infra for upgrading queries and models
  - Product-wide framework for opting into sharing with OpenAI

- Activation (side quest: Michael)
  - Michael to work with Growth to identify optimizations to getting started with Product Analytics

### High priority

TODO

### Low priority / side quests

TODO

**Comment by benjackwhite** [comment]
# Team Infra

**Support hero:** @frank

- Ben out for half of next sprint
- Public holidays as well

## Hang over items from previous sprint

1. 🟢 AWS cost usage exercise
- Done here https://github.com/PostHog/posthog-cloud-infra/issues/2658 
1. 🟡 Spontaneous look into AZ aware routing to try and reduce our biggest cost centre
1. 🔴 Purchase reserved RDS instances @frankh 
- why? Risk vs cost works out - to be fully confirmed with pipeline team
3. OpenVPN cert updates (only going to worry about CI/CD and infra peeps) @danielxnj 
- why? Current security risk as they have no expiry
4. Get app onto Canary deploys now that we are on contour @danielxnj 
- why? This is about having safer automatic deployments (based on HTTP errors) 
5. Test out capture-rs traffic with contour @frankh 
7. 🟡 Get flagged PoC for proper permissions out @benjackwhite 



1. 🟡 Rollout redirect approach for static assets @danielxnj
- Tested on dev and works, just needs more validation and slow rollouts
3. 🟢 More follow up on SOC2 and compliance with customers @danielxnj 
- Followed up with Simon and follow up in Drata around some missing bits around security networking etc.
- Docker image updates
4. 🟢 Get everything onto contour ingress so we can offer Pipeline a way of gradually rolling out capture
- This has taken forever but we are very nearly there (EU is fully on contour)
6. 🟢 Job spec out for SRE hire @benjackwhite 
7. 🟢 Attempt Mr Blobby v3 with EBS shared mount volume @benjackwhite  @danielxnj 
- Attempted but not an option - we're basically backing out of work here as Mr Blobby seems okay for now
- Postgres would be the next potential test scratch DB
9. 🟢 Get docs and JS fully ready for rollout of `.i.posthog.com` domains
- @benjackwhite to ping marketing and see if we want to announce anything


## OKR

1. 💪 Deploy with confidence 🟡
- 🟢 Manually controlling traffic between two versions of a service
- 🟢 Automatic canary deploying with rollbacks based on HTTP
- 🔴 Manual canary deploys for testing scary changes
1. 💰 Keep costs under control ✅
1. 👀 More Observability 🔴
1. 🎯 Soc2 ✅

### High priority


### Low priority / side quests
- Add alerts for ECR vulnerabilities @danielxnj

**Comment by EDsCODE** [comment]
# Team Data Shack

### OKR Q2 2024

### Objective

**Release data warehouse to everyone**

* **Key Results:**
    * Integration first experience
         * schemas are reliable
         * modeling of each integration is clear
         * Good automatic roll up views and joins
         * Wizard to onboard people
    * Establish a solid pattern to build integrations
    * Complete data warehouse experience in the rest of the app (insights, feature flags, experiments)

**James as a Service -> Clickhouse as a Service**

* **Key Results:**
    * Better Visibility
         * Regularly testing backups
         * Monitoring/alerting
         * Mutations
         * Moves
    * Management
         * Managing/killing mutations
    * Self Serve
         * Schema design feedback (James non blocking
         * Schema management   
    * Automation
         * Replace/Upgrade replicas
         * Upgrading to 24.04
         * Disk configs 
 

### Retro

**Product**
- [x] @EDsCODE Implement data warehouse series logic at hogql layer (vs query by query)
- [ ] @Gilbert09 property typings for data warehouse tables (for autocomplete)
- [x] Zendesk integration (helping @annikaschmid dogfood data warehouse product) -> should lead to adding row limits

**Infra**
- [ ] wire up ansible to pull inventory from TF or AWS using tags
- [ ] have US instance up with ansible playbook from galaxy
- [ ] setup CI for both of the above

### High priority

**Product**
- [ ] @EDsCODE High/medium priority fixes suggested by @annika here https://github.com/PostHog/posthog/issues/21105
- [ ] @Gilbert09 property typings for data warehouse tables (for autocomplete)
- [ ] @Gilbert09 zendesk integration v0.2. Support more granular sync selections when first setting up

**Infra**

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @jurajmajerik  
Days off:
Juraj:  0 days
Neil: 1 days

## Retro


- Experiments UI draft + results view - basic version with all pre-existing functionality we can release - @jurajmajerik -> Should be good to go by Friday
- Investigate and fix (or find alternatives) for cohort OOM errors for $bigClients - @neilkakkar -> in progress
- Add property filters to event matching in cohorts - @neilkakkar -> 🚫 
- Find a good flow for restricting users to exposed users in secondary metrics for experiments - @neilkakkar -> 🚫 


## Hang over items from previous sprint

---

## OKRs

1. 🔴 Doubling down on surveys
2. 🟡 Doubling down on experiments

### High priority

- Collect feedback on the new Experiments UI from our top users - @jurajmajerik 
- Write RFC for Surveys branching logic, create wireframes - @jurajmajerik 
- Fix Surveys bugs: [one](https://github.com/PostHog/posthog/issues/20851), [two](https://github.com/PostHog/posthog/issues/20866) - @jurajmajerik 
- Explore experiment metrics easily in insights - @jurajmajerik 
- Add property filters to event matching in cohorts - @neilkakkar
- Find a good flow for restricting users to exposed users in secondary metrics for experiments - @neilkakkar
- RFC for feature flags rewrite - @neilkakkar 


### Low priority / side quests

- Continue improving Experiments UI - links to docs, MAI calculator - @jurajmajerik 
- Temporal queues for feature success - @neilkakkar

**Comment by robbie-c** [comment]
# Team 🕸️ 
Off: none

## Last sprint 
* Fixed many bugs relating to the backfill, new completed 95% done on US, 50% on EU (which is much less)
* HogQL sessions table optimisation
* Sessions channel type fixes
* Wrote some docs for web analytics

## Hang over items from previous sprint
* Run sessions backfilll script
* Add session filters

## OKR
1. Finish the [beta to-do list](https://github.com/PostHog/posthog/issues/18547)
2. Find 5 high-ICP reference customers

### High priority
* Pause sessions table work (except nursing the backfill) to work on person-less events
* Person-less events
  * posthog-js changes
  * UI problem solving whack-a-mole
* Write more docs for web analytics (e.g. what is a Channel Type?)

### Low priority
* Session filters for all posthog events
* Add session properties to the property definition endpoint
* polish and be responsive to support

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>
<summary>Retro items</summary>
- [x] Team offsite!

### High priority
- @raquelmsmith 
    - [x] Rethink some activation metrics (remove team member invited, double check rates with diff activities)
    - [x] Personless events RFC
    - [x] Billing feature overrides
    - [x] re-grandfather unlimited projects and history view for existing customers
    - [x] Allow reads/writes to description without team collab feature https://github.com/PostHog/posthog/pull/21027
- @xrdt 
     - [x] Pricing calculator - cap slider at 250M, under final tier say contact us
          - How many people send us over 250M events per month but aren't on special plan
           - Customer with potential price differences here: https://docs.google.com/document/d/1fb6XzDNSOK0G4ICuEQhFtkfDczCIeK6ZNh1dtTA-vU4/edit?disco=AAABJGMkSRM
     - [x] Change stripe price for product analytics, highest tier 0.00001 - migrate people who aren't over that volume?
          - New price: https://dashboard.stripe.com/prices/price_1OtaGHEuIatRXSdzIzbyvCym
          - Run script compare_prices to see who would pay more
- @zlwaterfield 

</details>


## OKR
✅=finished 🟢=on track to finish this quarter 🟡=might not finish 🔴=won't finish
- 🟡 Improve activation rates for product analytics and session replay
  - ✅   Polish onboarding flow with Cory
  - 🟢  Run 1 experiment per week
  - 🟢  Do an average of 2 customer interviews per week
  - 🔴  Redefine and improve activation rates for PA & SR
    - Did not really get to this, revisiting for PA in Q2
- 🟢  Launch pricing for:
  - 🟢 Teams

## This sprint
### High priority
- @raquelmsmith 
    - [ ] Dashboard creation from onboarding
    - [ ] Feature gating for activity panel
      - Migrate customers
    - [ ] Feature gating for automatic provisioning
      - Migrate customers
    
- @xrdt 
    - Out for the first week of this sprint
    - [ ] Support billing for person processing addon
- @zlwaterfield 
    - [ ] RFC for simpler upgrading
    - [ ] Move the Growth 2.0 dashboard to a Notebook and finish it the charts we have data for
    - [ ] Helping Ian improve Next.js documentation
    - [ ] Continue email verification investigation
    - [ ] Review / test Michael’s dark mode PR: https://github.com/PostHog/posthog/pull/21149
    - [ ] Looking at some recurring Sentry bugs (e.g https://github.com/PostHog/billing/pull/499) 
    - Nice to haves:
      - CORS issue for Next.js reverse proxy
      - Billing as a product competition/market analysis and breakdown of possible features
    - Metrics to watch:
      - How many organizations sign up without a name: https://us.posthog.com/project/2/insights/SmImt4r7 (see if this is an indicator of lower chance to subscribe to paid plan)
      - Sign up benefits - generic, vs product analytics focused vs logs: https://us.posthog.com/project/2/experiments/24738

**Comment by tiina303** [comment]
# Team Pipeline

Off: Tomas (2 days)
Support: Tomas

## Retro

### High priority

- [ ] Person property updates don't slow down ingestion - Brett
   - sidetracked by other stuff, e.g. support
- [ ] Launch Person IDs on Events for everyone - Ted & Tomás
  - [ ] The squash job runs weekly on Sundays (*gated on team IDs because irreversible*)
        - **blocked on disk space issues**
  - [ ] We have metrics on override table size (total size, highest total per team) to correlate with query performance
  - [x] **v3 read path** with Analytics: overrides are loaded in a CH dictionary available for query, we monitor its resource usage (memory, cpu) as we backfill values
        - without the dictionary part as discussed that to be perf tested first
  - [x] Overrides for all teams are backfilled to CH and available for query
  - [ ] Read path feature flag rolled out to all teams
        - blocked by HogQL insights migration (to avoid incisistencies of data between insights that are not using HogQL yet)
- [x] Replay overflow implemented, plugin-server detection part - Xavier
    - Likely will ship by EOW

## OKR
✅=finished 🟢=on track to finish this quarter 🟡=might not finish 🔴=won't finish

- ✅ Wrapped up Webhooks Delivery system - Brett
- ✅ Wrapped up capture-rs posthog-js rollout - Xavier
- 🔴 Wrapped up New Pipeline UI - Tiina
    - side tracked by CDP, team leading, personless events
- 🟡 Wrapped up PoE (with backfills and squashes) - Ted
    - pending on other teams
- 🟢 Volume spikes on session replay do not hurt other users - Xavier
- 🔴 Person property updates don't slow down ingestion - Brett
    - got deprioritised over cloud to cloud migrations & deletions breaking
- 🔴 Requested data deletion is complete and reliably executed going forward - Tiina
    - turned out to be more complex - tracked in [issue](https://github.com/PostHog/posthog/issues/20221)

### High priority

- [ ] Personless (Tiina & everyone else)

### Low priority / side quests

- [ ] Helping out with HogQL migration (Ted)

**Comment by neilkakkar** [comment]
# Team Feature Successsss 🐍 

**Support hero:** @neilkakkar   
Days off:
Juraj:  1 days
Neil: 1 days

## Retro

- Collect feedback on the new Experiments UI from our top users - @jurajmajerik  ->  survey is out, some good feedback already, most people love it! Will potentially do some individual reaching out next sprint.
- Continue improving Experiments UI - @jurajmajerik 
    - no results empty state: show checklist for what events are missing. Explore button in the middle of empty state. ✅
    - banners for what to do next - ✅
    - data collection participants bug - set default to 20% for happy path, 50% for edges - no, will rollover. 
    - Tooltips (stretch)
    - link to docs (stretch)
    - investigate MAI calculator (stretch)
- Add property filters [and exact date matching] to event matching in cohorts - @neilkakkar -> ✅
- Find a good flow for restricting users to exposed users in secondary metrics for experiments - @neilkakkar -> 🔄
- RFC for feature flags rewrite - @neilkakkar -> ✅


## Hang over items from previous sprint

---

## OKRs

1. Make sure feature flags can handle 10x current scale
2. Polish new experiments UI & collect feedback
3. Add most requested surveys functionality

### High priority

- Write RFC for Surveys branching logic, create wireframes, and talk to YC & RG - @jurajmajerik 
- Get the surveys preview fix over the line https://github.com/PostHog/posthog/pull/20321 - @jurajmajerik 
- Data collection participants bug - set default to 20% for happy path, 50% for edges - @jurajmajerik 
- Explore experiment metrics easily in insights - @neilkakkar  
- Learn Rust, and minimal PoC of new feature flags service - @neilkakkar 

### Low priority / side quests

- Fix some Surveys bugs: [one](https://github.com/PostHog/posthog/issues/20851), [two](https://github.com/PostHog/posthog/issues/20866) - @jurajmajerik 
- Temporal queues for feature success - @neilkakkar 
- Setup instrumentation for flip-flopping problem of experiment significance - @neilkakkar
